### DISCUSSION TITLE
Where to load meta description from cms?

### DISCUSSION THREAD

**Discussion by enricoschaaf** [vercel/next.js#17379] [Help] [answered]
I want to add a global meta description for all pages from a CMS at build time while still keeping all pages statically optimized, does every page need a get static props now or can I use _app/_document for that without breaking any static optimization?

**Comment by jamesmosier** [comment, marked as answer]
There is currently no way to globally hydrate data into all components during build time using `getStaticProps`. Therefore you'd need to hydrate them in each page using `getStaticProps` and `next/head`.

**Reply by enricoschaaf** [reply]
Thanks😔☺️
